### DISCUSSION TITLE
Is it possible to connect geoserver wms and deck.gl by getCapabiliites?

### DISCUSSION THREAD

**Discussion by Woo-Yeol** [visgl/deck.gl#7393] [Q&A] [answered]
Hi I'm student developer.

 I use React with Typescript, deck.gl and GeoServer.

I want to show GeoServer's Tile Layer in my platform with wms/wmts protocol.

And TileLayer requires extend and layer name to establish url.

I think getCapabillites can offer that info.

But I cannot find how to use WMS 1.1.1 getCapabillites in deck.gl.

Is it Impossible to use getCapabillites?

**Comment by radoslavirha** [comment, marked as answer]
Hi,

the key to WMS layers is to use `getTileData`  in TileLayer where you'll build URL per tile.

My way of implementing WMS is to call `getCapabillites` in administration (where admin provides base URL), save selected params (format, layers,....) and build TileLayer with those params.

I think it's outside of deck.gl's scope to implement `getCapabillites`, It expects you have those params and you use it in TileLayer.

Example:
https://github.com/visgl/deck.gl/discussions/6885#discussioncomment-2703052

**Reply by Woo-Yeol** [reply]
Thanks for your answer!

I will parsing data from `getCapabillites` with another tool because it is xml based...

**Reply by radoslavirha** [reply]
Just use https://www.npmjs.com/package/wms-capabilities

**Reply by Woo-Yeol** [reply]
Thanks!

Now I solved the problem and It works well.

**Reply by Pueablo** [reply]
> ¡Gracias!
> 
> Ahora resolví el problema y funciona bien.

Hi, Can you help me? I have the same problem, please!

**Reply by ibgreen** [reply]
This is probably not sufficient, but just in case, loaders.gl 3.3 was just published and it includes a new module `@loaders.gl/wms` with a parser for the WMS GetCapabilities XML response in case that is helpful. Unfortunately the docs are not up yet. but there is some info on github: https://github.com/visgl/loaders.gl/blob/master/modules/wms/docs/formats/wms.md.

**Reply by Pueablo** [reply]
thanks!
